services/src/main/java/org/keycloak/protocol/oid4vc/issuance/mappers/OID4VCIssuedAtTimeClaimMapper.java

```diff
@@ -0,0 +1,141 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.protocol.oid4vc.issuance.mappers;
+
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.UserSessionModel;
+import org.keycloak.protocol.ProtocolMapper;
+import org.keycloak.protocol.oid4vc.model.CredentialSubject;
+import org.keycloak.protocol.oid4vc.model.VerifiableCredential;
+import org.keycloak.provider.ProviderConfigProperty;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * Map issuance date to the credential, under the default claim name "iat"
+ * <p>
+ * subjectProperty can be used to change the claim name.
+ * <p>
+ * Source of the information can either be computed, or read from the VerifiableCredential object
+ * bearing other claims. Default is the value in the verifiable credential.
+ * <p>
+ * We will use the java.time.temporal.ChronoUnit enum values to help flatten down the time.
+ *
+ * @author <a href="mailto:francis.pouatcha@adorsys.com">Francis Pouatcha</a>
+ */
+public class OID4VCIssuedAtTimeClaimMapper extends OID4VCMapper {
+
+    public static final String MAPPER_ID = "oid4vc-issued-at-time-claim-mapper";
+
+    // Omit value if defaults to "iat"
+    public static final String SUBJECT_PROPERTY_CONFIG_KEY = "subjectProperty";
+
+    // We will use the java.time.temporal.ChronoUnit enum values to help flatten down the time.
+    // Omit property if no truncation.
+    public static final String TRUNCATE_TO_TIME_UNIT_KEY = "truncateToTimeUnit";
+
+    // Time computed (COMPUTE) or taken from the verifiable credential (VC).
+    // Defaults to VC. Falls back to COMPUTE.
+    public static final String VALUE_SOURCE = "valueSource";
+
+    private static final List<ProviderConfigProperty> CONFIG_PROPERTIES = new ArrayList<>();
+
+    static {
+        ProviderConfigProperty subjectPropertyNameConfig = new ProviderConfigProperty();
+        subjectPropertyNameConfig.setName(SUBJECT_PROPERTY_CONFIG_KEY);
+        subjectPropertyNameConfig.setLabel("Issuance Date Claim Property Name");
+        subjectPropertyNameConfig.setHelpText("Name of the property to contain the iat.");
+        subjectPropertyNameConfig.setType(ProviderConfigProperty.STRING_TYPE);
+        CONFIG_PROPERTIES.add(subjectPropertyNameConfig);
+
+        ProviderConfigProperty truncateToTimeUnit = new ProviderConfigProperty();
+        truncateToTimeUnit.setName(TRUNCATE_TO_TIME_UNIT_KEY);
+        truncateToTimeUnit.setLabel("Truncate To Time Unit");
+        truncateToTimeUnit.setHelpText("Truncate to time unit");
```

Comment:
Same as other options, is it possible to make this one a bit more friendly for the administrator? I can imagine the `helpText` can be improved to make it more descriptive what the individual values means? Also do we really need everything from the `ChronoUnit` enum? Some other places in Keycloak (like EG. when yo go to `Realm Settings` and subtab `Tokens` in the admin console) use time units like `Seconds`, `Minutes`, `Hours` or `Days`. If we limit it, it can be used the type `LIST_TYPE` where admin can choose from the available values, which is certainly more friendly than the requirement of typing the value by hand. But maybe I am missing if there are really use-cases for the formats like `NANOS` or `MICROS` for example? WDYT?